### PR TITLE
backport: exp/slices.sortFunc to "slices"

### PR DESCRIPTION
**Pull request by Nasfame**
backport: slices.SortFunc from exp/slices

For #61374

### CODE REVIEW

File: src/cmp/cmp.go [hunk 1]
```diff
@@ -31,9 +31,11 @@ func Less[T Ordered](x, y T) bool {
 
 // Compare returns
 //
-//	-1 if x is less than y,
-//	 0 if x equals y,
-//	+1 if x is greater than y.
+// -1 if x is less than y,
```
Comment (1268) [inline src/cmp/cmp.go:34]: this is not correct - edge case of `go fmt`. best fix would probably be to move the next line left one space relative to its original position

File: src/cmp/cmp.go [hunk 2]
```diff
@@ -31,9 +31,11 @@ func Less[T Ordered](x, y T) bool {
...
+//
+//	0 if x equals y,
+//
+// +1 if x is greater than y.
```
Comment (1268) [inline src/cmp/cmp.go:38]: this is not correct - edge case of `go fmt`. best fix would probably be to move the next line left one space relative to its original position

File: src/slices/sort.go [hunk 3]
```diff
@@ -28,6 +28,18 @@ func SortFunc[S ~[]E, E any](x S, cmp func(a, b E) int) {
...
+// function. This sort is not guaranteed to be stable.
+// cmp(a, b) should return a boolean i.e a < b, true when
+// a > b and false when a <= b
+// SortFuncBasedOnBool requires that cmp is a strict weak ordering.
+// See https://en.wikipedia.org/wiki/Weak_ordering#Strict_weak_orderings.
+// Backported from https://pkg.go.dev/golang.org/x/exp/slices#SortFunc
+func SortFuncBasedOnBool[S ~[]E, E any](x S, cmp func(a, b E) bool) {
+	n := len(x)
+	pdqsortCmpFuncBasedonBool(x, 0, n, bits.Len(uint(n)), cmp)
+}
```
Comment (1268) [inline src/slices/sort.go:41]: uh what is this? I like the idea, but I believe the plan was just to simply remove the bool version totally. personally I prefer the bool version, but if its going into the standard library is probably needs a better name

File: src/slices/zsortanyfunc.go [hunk 4]
```diff
@@ -1,3 +1,63 @@
...
+ * Copyright 2023 The Go Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file.
+ */
+
+/*
+ * Copyright 2023 The Go Authors. All rights reserved.
+ * Use of this source code is governed by a BSD-style
+ * license that can be found in the LICENSE file.
+ */
```
Comment (1268) [inline src/slices/zsortanyfunc.go:59]: trash.

File: src/cmp/cmp.go (hunk 2 above)
Comment (Nasfame) [reply src/cmp/cmp.go:38]: Please don’t reply on this GitHub thread. Visit [golang.org/cl/511395](https://go-review.googlesource.com/c/go/+/511395#message-38c5f335c9a6a4c1e10357c82a8855cdc41bffa2).

File: src/slices/sort.go (hunk 3 above)
Comment (Nasfame) [reply src/slices/sort.go:41]: Please don’t reply on this GitHub thread. Visit [golang.org/cl/511395](https://go-review.googlesource.com/c/go/+/511395#message-38c5f335c9a6a4c1e10357c82a8855cdc41bffa2).

File: src/cmp/cmp.go (hunk 2 above)
Comment (1268) [reply src/cmp/cmp.go:38]: nope, I am not doing that. I am signed in here, not there, so if you cant accept that then feel free to ignore my comments.

File: src/slices/sort.go (hunk 3 above)
Comment (1268) [reply src/slices/sort.go:41]: nope, I am not doing that. I am signed in here, not there, so if you cant accept that then feel free to ignore my comments.

File: src/cmp/cmp.go (hunk 2 above)
Comment (Nasfame) [reply src/cmp/cmp.go:38]: @1268 not sure I m using Goland, and the doc of the function is being rendered as "-1" is rendered as 

- 1

Comment (1268) [reply src/cmp/cmp.go:38]: here is correct code:
https://go.dev/play/p/t0cnJA_mbIc

Comment (Nasfame) [reply src/cmp/cmp.go:38]: image

Comment (1268) [reply src/cmp/cmp.go]: Its not my job to troubleshoot problems with your editor. only to direct you to what the SPEC says:

https://go.dev/doc/comment

so regardless of any output of any editor, the code I have written is correct according to the spec.

**Pull request closed without merging**